src/Architectures.jl

```diff
@@ -120,5 +127,11 @@ end
 @inline convert_args(::GPU, args) = CUDA.cudaconvert(args)
 @inline convert_args(::GPU, args::Tuple) = map(CUDA.cudaconvert, args)
 
+# Deprecated functions
+function arch_array(arch, arr) 
+    @warn "`arch_array` is deprecated. Use `on_architecture` instead."
+    return on_architecture(arch, arr)
+end
+
```

Comment:
Do we really need this here? What for? The PR removes it from everywhere.
Do we keep it here in case people were doing

```Julia
using Oceananigans.Architectures: arch_array
```

in their code?

I suggest we remove it and if we want to inform people we bump a patch release and mention it in the release notes? (No need to bump a minor release since this is not a public API change.)


```suggestion
```

Reply:
OK, I see `arch_array` being used, e.g., over at ClimaOcean:

https://github.com/CliMA/ClimaOcean.jl/blob/a0e0fc0b1cfb3a637bb8f234bf558e099038c6cb/src/NearGlobalSimulations/one_degree_global_simulation.jl#L76

Let's remove it and bump a patch release.

Reply:
I liked the idea of keeping a list of deprecated functions that will be removed after a couple of releases. 
At least users know what has happened and are ready for the change that will happen down the line. 
I would go with the majority of the opinions on this one.

Reply:
OK keep it! No problemo! True, we can remove it after 1-2 breaking releases.